lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -70,7 +70,11 @@ def discover
           numeric_id = NUMBERS_RE =~ numeric_id ? numeric_id.to_i : nil
 
           log.debug "synapse: discovered backend #{name} at #{host}:#{server_port} for service #{@name}"
-          new_backends << { 'name' => name, 'host' => host, 'port' => server_port, 'id' => numeric_id}
+          new_backend = { 'name' => name, 'host' => host, 'port' => server_port, 'id' => numeric_id}
+          if !weight.nil?
```

Comment:
Should we keep the handling with name consistent (i.e. name will just be nil if it doesn't have a name, same for weight)?